quint/src/runtime/trace.ts

```diff
@@ -336,23 +344,28 @@ class TraceRecorderImpl implements TraceRecorder {
       }
     }
 
-    const notOk = fromResult(outcome)
-    const prevNotOk = fromResult(this.bestTrace.result)
-
-    // Prefer short traces for error, and longer traces for non error.
-    // Therefore, override the best trace only if:
-    //  - there is an error, the new trace is shorter, or the old trace is non-error;
-    //  - there is no error, the new trace is longer, and there was no error before.
-    const override = notOk
-      ? this.bestTrace.args.length === 0 || !prevNotOk || this.bestTrace.args.length >= bottom.args.length
-      : !prevNotOk && this.bestTrace.args.length <= bottom.args.length
-
-    if (override) {
-      this.bestTrace = bottom
-      this.bestTraceSeed = this.runSeed
-      this.bestTrace.result = outcome
-      this.bestTrace.args = trace
-    }
+    return this.traces.sort((a, b) => {
+      // Prefer short traces for error, and longer traces for non error.
+      // Therefore, trace a is better than trace b if
```

Comment:
comment doesn't describe the behaviour correctly afaict

if aOk & bNotOk, then b is better (because errors are preferred), but the comment says "trace a is better than trace b if {}, {}, or b has an error"

Reply:
You're right, thanks! Fixed it.

(I thought I had merged this already, completely forgot that this fix was pending, sorry for the delay!)